### PR TITLE
Add spending limit notifications

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Description
This PR add the basic notification to signal reaching of spending limits as described in https://github.com/gitpod-io/gitpod/issues/11404. The static message is provided by the server API.

**Screenshot** shows a notification when a spending limit of **1** is reached.

The notification is dismissible, but will reappear on reload until the condition is resolved. 

Screen Shot 2022-07-22 at 14 51 36

## Related Issue(s)

Relates to #11404

## How to test
* setting up proper refresh rate for `usage` component
* setting up a spending limit on a team with billing enabled
* starting a workspace and waiting for the usage to reach 80% or more
* reloading the dashboard

## Release Notes

```release-note
Notify when spending limit is reached.
```

## Werft options:

- [x] /werft with-preview
- [x] /werft with-payment

### CODE REVIEW

**Comment by roboquat** [comment]
@AlexTugarev: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/server/ee/src/workspace/gitpod-server-impl.ts [hunk 1]
```diff
@@ -2096,6 +2096,26 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         });
     }
 
+    async getNotifications(ctx: TraceContext): Promise<string[]> {
+        const result = await super.getNotifications(ctx);
+        const user = this.checkAndBlockUser("getNotifications");
+        if (user.usageAttributionId) {
```
Comment (geropl) [inline components/server/ee/src/workspace/gitpod-server-impl.ts:2102]: This has to be replaced by the `getBillingMode(user)` method. But as it's not there it, this is fine.

File: components/server/ee/src/workspace/gitpod-server-impl.ts [hunk 2]
```diff
@@ -2096,6 +2096,26 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
...
+            const allSessions = await this.listBilledUsage(ctx, user.usageAttributionId);
+            const totalUsage = allSessions.map((s) => s.credits).reduce((a, b) => a + b, 0);
```
Comment (geropl) [inline components/server/ee/src/workspace/gitpod-server-impl.ts:2104]: Ideally we'd get that from the service as well, but we can replace that once we do.

**Review by geropl — APPROVED** [review]
LGTM, thank you! :pray:

**Pull request merged**